instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java

```diff
@@ -32,14 +33,16 @@ public Span startSpan(Class<?> target, Method method) {
     // We create span and immediately update its name
     // We do that in order to reuse logic inside updateSpanNames method, which is used externally as
     // well.
-    Span span = tracer.spanBuilder("jax-rs.request").startSpan();
-    updateSpanNames(span, BaseTracer.getCurrentServerSpan(), target, method);
+    Context context = Context.current();
+    Span span = tracer.spanBuilder("jax-rs.request").setParent(context).startSpan();
+    updateSpanNames(context, span, BaseTracer.getCurrentServerSpan(context), target, method);
     return span;
   }
 
-  public void updateSpanNames(Span span, Span serverSpan, Class<?> target, Method method) {
-    // When jax-rs is the root, we want to name using the path, otherwise use the class/method.
-    String pathBasedSpanName = getPathSpanName(target, method);
+  public void updateSpanNames(
+      Context context, Span span, Span serverSpan, Class<?> target, Method method) {
+    String pathBasedSpanName =
```

Comment:
Here also I suspect `updateSpanName(serverSpan)` isn't doing anything, and for the non-serverspan case we don't need the `getApplicationRoot` call

Reply:
I think this is needed, I added a test for this now